person.rb

```diff
@@ -1,27 +1,31 @@
 require './nameable.rb'
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
   attr_accessor :name, :age
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
-  private
-
-  def of_age?
-    true if @age > 18
-  end
-
-  public
-
   def can_use_services?
     of_age?
   end
 
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals << rental
+    rental.book = self
+  end
```

Comment:
- [ ] The same thing with the `add_rentals` method in the Person class. The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
end
```